### PR TITLE
Standardise CI and Makefile across repositories

### PR DESCRIPTION
**Pull request by phklive**
In this PR I propose a standardisation of the CI and Makefile across Miden repositories. 

With the following changes: 
- Cargo make to Make
- CI using Cargo make to Make

Once this first one is approved I will port it to the following repos:
- VM
- Crypto
- Client
- Compiler

Closes: #363

### CODE REVIEW

**Review by hackaugusto — APPROVED** [review]
LGTM, thanks :)

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+# --- installing --------------------------------------------------------------------------------
+
+.PHONY: install-node
+install-node: ## Installs node with testing feature enabled
+	cargo install --features testing --path bin/node
+
+.PHONY: install-faucet
+install-faucet: ## Installs faucet with testing feature enabled
+	cargo install --features testing --path bin/faucet
```
Comment (bobbinth) [inline Makefile:61]: Should we have 2 versions of these commands? One for installing with `testing` feature and one without?

**Review by bobbinth — APPROVED** [review]
Looks good! Thank you! I left one comment inline. Once this is addressed, we can merge.

File: Makefile (hunk 1 above)
Comment (phklive) [reply Makefile:61]: Sure.

**Pull request merged**